pkg/segment/results/mresults/metricresults.go

```diff
@@ -659,6 +659,23 @@ func (r *MetricsResult) FetchPromqlMetricsForUi(mQuery *structs.MetricsQuery, pq
 			allTimestamps[ts] = struct{}{}
 		}
 	}
+	//add timestamps where metrics value is null
+	if calculatedInterval >= 600 {
+		t := time.Unix(int64(startTime), 0).UTC()
+		truncated := t.Truncate(time.Hour)
+		if t.Before(truncated) {
+			truncated = truncated.Add(time.Hour)
+		}
+		// Convert back to epoch time
+		roundedDownEpochTime := truncated.Unix()
+		startTime = uint32(roundedDownEpochTime)
+	}
```

Comment:
I'm missing something:
1. Why do we only do this when the interval time is at least 5 minutes?
2. What is the purpose of the `if` on 666-668? Won't the condition always be false? Or is it true when `startTime` is exactly on an hour and that's the point of it?

Reply:
Modified the code on 667 to conditionally move back start time by 1hr/5min/15min/1min to display time range correctly